src/engine/source/router/src/orchestrator.cpp

```diff
@@ -0,0 +1,548 @@
+#include <router/orchestrator.hpp>
+
+#include "entryConverter.hpp"
+#include "worker.hpp"
+
+namespace router
+{
+
+namespace
+{
+/**
+ * @brief Validates that the pointer is not empty
+ */
+template<typename T>
+void validatePointer(const T& ptr, const std::string& name)
+{
+    if constexpr (std::is_same_v<T, std::weak_ptr<typename T::element_type>>)
+    {
+        if (ptr.expired())
+            throw std::runtime_error {"Configuration error: " + name + " cannot be empty"};
+    }
+    else if (!ptr)
+        throw std::runtime_error {"Configuration error: " + name + " cannot be empty"};
+}
+} // namespace
+
+// Private
+base::OptError Orchestrator::forEachWorker(const WorkerOp& f)
+{
+    for (const auto& worker : m_workers)
+    {
+        if (auto error = f(worker); error)
+        {
+            return error;
+        }
+    }
+    return std::nullopt;
+}
+
+/**************************************************************************
+ * Manage configuration - Dump
+ *************************************************************************/
+base::OptError loadTesterOnWorker(const std::vector<EntryConverter>& entries, const std::shared_ptr<IWorker>& worker)
+{
+    for (const auto& entry : entries)
+    {
+        auto err = worker->getTester()->addEntry(test::EntryPost(entry), true);
+        if (err)
+        {
+            return err;
+        }
+        worker->getTester()->updateLastUsed(entry.name(), entry.lastUse().value_or(0));
+        worker->getTester()->enableEntry(entry.name());
+    }
+    return std::nullopt;
+}
+
+base::OptError loadRouterOnWoker(const std::vector<EntryConverter>& entries, const std::shared_ptr<IWorker>& worker)
+{
+    for (const auto& entry : entries)
+    {
+        auto err = worker->getRouter()->addEntry(prod::EntryPost(entry), true);
+        if (err)
+        {
+            return err;
+        }
+        worker->getRouter()->enableEntry(entry.name());
+    }
+    return std::nullopt;
+}
+
+void saveConfig(const std::weak_ptr<store::IStoreInternal>& wStore, const base::Name& storeName, const json::Json& dump)
```

Comment:
It is not necessary, if it fails it will be logged in the log, it is not a problem of the API interface but a functionality of the module in general. The execution result of each method that calls it is independent of whether or not it was able to save the configuration